src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -956,13 +957,46 @@ public bool StartsWith(string value)
             return StartsWith(value, StringComparison.CurrentCulture);
         }
 
+        // A hint for the inliner to always prescan this method
+        // and take the constant input path
+        [Intrinsic]
         public bool StartsWith(string value, StringComparison comparisonType)
         {
             if (value is null)
             {
                 throw new ArgumentNullException(nameof(value));
             }
 
+#if TARGET_64BIT && CORECLR && !BIGENDIAN
```

Comment:
This will work for AOT and NativeAOT fine, and you are just trying to exclude this for Mono, correct? 

The ifdef should be !MONO then.

Reply:
Yes, was trying to avoid potential leftovers and throughput impact for mono since IsKnownConsant is not yet implemented there